### PR TITLE
Suggestion: add default dlint.yaml

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
While I think we should have some documentation for this on [docs.daml.com](https://docs.daml.com) and ideally some way to query the dlint engine itself (something like `daml dlint --list-rules` maybe?) to get a list of the rules in the current version, I believe another nice way to make our user's life a bit easier is to include a default dlint config file in each template, that lists all of the available rules.

@cocreature @shayne-fletcher-da This is a quick draft based on Slack comments; if the idea seems valid I'm happy to flesh this out to actually list all of the options (though I'll need a quick pointer to where I can find them).

### CODE REVIEW

File: templates/default-dlint-yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Warns about uses of snake_case instead of camelCase
+# Enabled by default. Uncomment to disable.
```
Comment (cocreature) [inline templates/default-dlint-yaml]: Side note: I wonder if we should disable this by default given how few users seem to like this hint.

File: templates/BUILD.bazel [hunk 2]
```diff
@@ -24,6 +24,7 @@ genrule(
             mkdir -p $$OUT/$$d
             cp -rL $$SRC/$$d/* $$OUT/$$d/
             cp $$SRC/default-gitignore $$OUT/$$d/.gitignore
+            cp $$SRC/default-daml-yaml $$OUT/$$d/.dlint.yaml
```
Comment (cocreature) [inline templates/BUILD.bazel]:
```suggestion
            cp $$SRC/default-dlint-yaml $$OUT/$$d/.dlint.yaml
```
I believe you also need to add it to `srcs`?

File: templates/default-dlint-yaml (hunk 1 above)
Comment (garyverhaegen-da) [reply templates/default-dlint-yaml]: Science says [underscores are better](https://ieeexplore.ieee.org/document/5521745). [[PDF](http://www.cs.kent.edu/~jmaletic/papers/ICPC2010-CamelCaseUnderScoreClouds.pdf)] 🙄

**Comment by garyverhaegen-da** [comment]
After discussion with @shayne-fletcher-da, there doesn't seem to be an easy way to get "all available rules", so I'm punting on that for the time being. I still believe this has value in its current state.

**Comment by hurryabit** [comment]
There's `hlint default` IIRC. Can we leverage something like this?

**Comment by shayne-fletcher-da** [comment]
> There's `hlint default` IIRC. Can we leverage something like this?

IIUC, the way that works is, you can run hlint over an existing project and it will produce a `.hlint.yaml` file that adds directives to ignore any of the warnings found in the project in that state. Yes, we can hook into that at some point.

**Pull request merged**